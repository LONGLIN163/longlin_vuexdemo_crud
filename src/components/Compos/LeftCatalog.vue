<template>
    <el-aside width="200px" style="background-color: rgb(238, 241, 246)">
      <el-menu 
        :default-active="activeIndex" 
        class="el-menu-demo" 
        mode="vertical" 
        @select="handleSelect"
      >
          <el-menu-item index='0' :key="0">All</el-menu-item>
          <el-menu-item v-for="(item,index) in types" :index="(index+1).toString()" :key="item+index">{{item}}</el-menu-item>
      </el-menu>
    </el-aside>
</template>

<script>
    import store  from '@/vuex/store'
    import {mapState,mapMutations} from 'vuex'
    export default {
        data() {
          return {
            activeIndex:'0'
          }
        },
        computed:{
          ...mapState(['types'])
        },
        methods:{

        handleSelect(key) {
          //console.log("key---",key);
          this.getListByType(this.types[key-1])
        },
          ...mapMutations(['getListByType'])
        },
        store
    }
</script>

<style scoped>
  .el-aside {
    color: #333;
  }
</style>