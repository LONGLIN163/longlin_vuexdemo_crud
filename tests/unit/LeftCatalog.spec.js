import { mount,createLocalVue} from '@vue/test-utils'
import LeftCatalog from '@/components/Compos/LeftCatalog'
import Vuex from 'vuex'

import ElementUI from "element-ui"

const localVue = createLocalVue();
localVue.use(ElementUI);  // install ElementUI to the local vue instance
localVue.use(Vuex); // install vuex to the local vue instance

let wrapper;
describe('Component LeftCatalog tests:', () => {

  // define them outside of breforeEach
  let actions
  let state
  let store

  // mock handleSelect()
  const handleSelect = jest.spyOn(LeftCatalog.methods, 'handleSelect').mockImplementation(() => {
    // we don't need have to mock mutations here
  })

  beforeEach(() => {

    const mockTypes=["Refined", "Unbranded", "Licensed", "Small", "Sleek", "Awesome"]

    // mock actions
    actions = {
      setArticleListAction: jest.fn().mockImplementation(() => {
        store.state.types=mockTypes
      })
    }

    //mock state
    state={
      articleList: [],
      types:[]
    }

    // create a fake store
    store = new Vuex.Store({
      state,
      actions
    })

    wrapper = mount(LeftCatalog,{
        localVue,
        store
    })
  })


  it('Get LeftCatalog snapshot and check init action has been called', async () => {
    expect(actions.setArticleListAction).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })
    
  it('test if handleSelect has been called',async () => {
    const selectedItem=await wrapper.findAll('[data-test="leftNav"]').at(1);
    await selectedItem.trigger('click')
    expect(handleSelect).toHaveBeenCalled()
    // check active status after being called
    expect(selectedItem.classes()).toContain('is-active')
  })

  afterEach(() => {
    wrapper.destroy();
  });


})