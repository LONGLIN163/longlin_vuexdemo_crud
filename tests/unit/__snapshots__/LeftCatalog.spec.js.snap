// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LeftCatalog tests: Get LeftCatalog snapshot and check init action has been called 1`] = `
<aside class="el-aside" style="width: 200px;">
  <div id="tasteTypes">Flavours</div>
  <transition-stub mode="out-in" class="el-menu-demo">
    <ul role="menubar" class="el-menu">
      <li role="menuitem" tabindex="-1" class="el-menu-item is-active" style="padding-left: 20px;">All</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Refined</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Unbranded</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Licensed</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Small</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Sleek</li>
      <li role="menuitem" tabindex="-1" class="el-menu-item" style="padding-left: 20px;" data-test="leftNav">Awesome</li>
    </ul>
  </transition-stub>
</aside>
`;
